plugins/signalmonitor/signalhistorydelegate.cpp

```diff
@@ -95,10 +95,15 @@ void SignalHistoryDelegate::paint(QPainter *painter, const QStyleOptionViewItem
 
     foreach (qint64 ev, events) {
         const qint64 ts = SignalHistoryModel::timestamp(ev);
-        if (ts >= startTime && ts < endTime) {
-            const int x = x0 + dx * (ts - startTime) / interval;
-            lines << QLine(x, y0 + 1, x, y0 + dy - 2);
-        }
+
```

Comment:
Ho you highlighted the timestamps checks... sorry. I tough it was about the drawLine / drawLines.
This change is minor and very specific while you scroll and the history length is wide.
Before it would continue reading past the endTime events, while now it break when it reach the first past endTime event, so no, it does not exactly do the same thing as before.